R/preprocess.R

```diff
@@ -0,0 +1,69 @@
+#' Add columns for interval censoring of primary and secondary events
+#'
+#' @param linelist ...
+#' @param ptime_lwr ...
+#' @param ptime_upr ...
+#' @param pwindow ...
+#' @param stime_lwr ...
+#' @param stime_upr ...
+#' @param swindow ...
+#' @family preprocess
+#' @autoglobal
+#' @export
+add_event_vars <- function(
+  linelist, ptime_lwr = NULL, ptime_upr = NULL, pwindow = NULL,
+  stime_lwr = NULL, stime_upr = NULL, swindow = NULL
+) {
+  linelist <- .rename_column(linelist, "ptime_lwr", ptime_lwr)
+  linelist <- .rename_column(linelist, "ptime_upr", ptime_upr)
+  linelist <- .rename_column(linelist, "stime_lwr", stime_lwr)
+  linelist <- .rename_column(linelist, "stime_upr", stime_upr)
+  linelist <- .rename_column(linelist, "pwindow", pwindow)
+  linelist <- .rename_column(linelist, "swindow", swindow)
+
+  if (is.numeric(pwindow)) {
+    cli::cli_warn("Overwriting using numeric value(s) of pwindow provided!")
+    linelist$pwindow <- pwindow
+  }
+
+  if (is.numeric(swindow)) {
+    cli::cli_warn("Overwriting using numeric value(s) of swindow provided!")
+    linelist$swindow <- swindow
+  }
+
+  if (is.null(stime_upr)) {
+    linelist <- mutate(linelist, stime_upr = stime_lwr + swindow)
```

Comment:
Performance doesn't matter hugely but noting as written this is a very inefficient function as every rename, mutate, etc all here leads to a data copy.